src/collection.js

```diff
@@ -354,17 +354,54 @@ export default class Collection {
     if (!Number.isInteger(at) || at <= 0) {
       throw new Error("Invalid argument, expected a positive integer.");
     }
-    // TODO: we process history changes forward, while it would probably be more
-    // efficient and accurate to process them backward.
-    return this.bucket
-      .listHistory({
-        pages: Infinity, // all pages up to target timestamp are required
-        sort: "-target.data.last_modified",
-        filters: {
-          resource_name: "record",
-          collection_id: this.name,
-          "max_target.data.last_modified": String(at),
-        },
+    let oldestRecord;
+    // First, check that the history plugin covers the entire range back to the
+    // requested timestamp.
+    // 1. find the record timestamp in the current collection which
+    // last_modified is immediately fresher than the requested timestamp.
```

Comment:
I realize that what we'd need is a `created_at` field, not `last_modified`. As is, this doesn't make any sense :(